### PR TITLE
pb-1916: Add changes for delete status update.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What this PR does / why we need it**:
```
    pb-1916: Add changes for delete status update.

        - minor cleanups in the job option parameter input function.
        - added backup object name and uid as label in the delete job
          and volumebackupdelete cr.
 ```
**Which issue(s) this PR fixes** (optional)
Closes # pb-1916

**Special notes for your reviewer**:
**Testing:**
**VolumeInfo list in the backupobject with failed status for kdmp:**
```
"volumes" : [
      {
        "name" : "pvc-3d183534-3454-4661-a8d0-d12bceb3da3f",
        "namespace" : "proxy",
        "pvc" : "nfs-data-cassandra-0",
        "backupId" : "445dc26034c6123a4a4c28b75cd8d92d",
        "status" : {
          "status" : "Failed",
          "reason" : "repository [generic-backup/proxy-proxy/] connect failed: failed to run the kopia connect command: exit status 1 stdout:  stderr: kopia: error: error connecting to repository: repository not initialized in the provided storage, try --help\n"
        },
        "driverName" : "kdmp",
        "storageClass" : "portworx-proxy-volume-volume0"
      },
      {
        "storageClass" : "portworx-proxy-volume-volume1",
        "name" : "pvc-3895c057-76f6-458a-8dc2-0da628d61b37",
        "namespace" : "proxy",
        "pvc" : "nfs-data-cassandra-1",
        "backupId" : "2a9d986e9ac5c288c05cd7f7ec5d75b3",
        "status" : {
          "status" : "Failed",
          "reason" : "repository [generic-backup/proxy-proxy/] connect failed: failed to run the kopia connect command: exit status 1 stdout:  stderr: kopia: error: error connecting to repository: repository not initialized in the provided storage, try --help\n"
        },
        "driverName" : "kdmp"
      },
      {
        "driverName" : "kdmp",
        "storageClass" : "portworx-proxy-volume-volume2",
        "name" : "pvc-aa7a5276-f481-4a64-9d44-056b231b8922",
        "namespace" : "proxy",
        "pvc" : "nfs-data-cassandra-2",
        "backupId" : "3724523ce6cfeec6f77436fe14f18dad",
        "status" : {
          "status" : "Failed",
          "reason" : "repository [generic-backup/proxy-proxy/] connect failed: failed to run the kopia connect command: exit status 1 stdout:  stderr: kopia: error: error connecting to repository: repository not initialized in the provided storage, try --help\n"
        }
      }
    ],
```

**Changed the naming convention for the delete job volumebackupdelete CR and secret. It is now <prefix of d>-<shortuid of backupobj>-<shortuid of snapshot>**
```
[root@central-valiant-howler-0 ~]# kubectl get jobs  -n px-backup  -l "backup-object-name=kdmp1"
NAME                COMPLETIONS   DURATION   AGE
d-dfb5046-2a9d986   0/1           12s        12s
d-dfb5046-3724523   0/1           12s        12s
d-dfb5046-445dc26   0/1           12s        12s
[root@central-valiant-howler-0 ~]# kubectl get volumebackupdelete  -n px-backup  -l "backup-object-name=kdmp1"
NAME                AGE
d-dfb5046-2a9d986   24s
d-dfb5046-3724523   24s
d-dfb5046-445dc26   24s
[root@central-valiant-howler-0 ~]# kubectl get secret  -n px-backup  -l "backup-object-name=kdmp1"
NAME                TYPE     DATA   AGE
d-dfb5046-2a9d986   Opaque   8      31s
d-dfb5046-3724523   Opaque   8      31s
d-dfb5046-445dc26   Opaque   8      31s
[root@central-valiant-howler-0 ~]#
```
Also added the backup object name and uid as labels in these resource for filtering.
TODO: trimming for the label is pending. Planning to take that clean up together in other place as well.
```
[root@central-valiant-howler-0 ~]# kubectl get volumebackupdelete d-dfb5046-445dc26 --show-labels -n px-backup
NAME                AGE   LABELS
d-dfb5046-445dc26   42s   backup-object-name=kdmp1,backup-object-uid=dfb50469-39bd-44eb-8098-7f644ea91531
[root@central-valiant-howler-0 ~]#
```
**Sample volumebackupdelete CR with failure status:**
```
[root@central-valiant-howler-0 ~]# kubectl get volumebackupdelete d-dfb5046-2a9d986 -n px-backup -o yaml
apiVersion: kdmp.portworx.com/v1alpha1
kind: VolumeBackupDelete
metadata:
  annotations:
    stork.libopenstorage.org/skip-resource: "true"
  creationTimestamp: "2021-09-30T02:13:03Z"
  generation: 2
  labels:
    backup-object-name: kdmp1
    backup-object-uid: dfb50469-39bd-44eb-8098-7f644ea91531
  managedFields:
  - apiVersion: kdmp.portworx.com/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:stork.libopenstorage.org/skip-resource: {}
        f:labels:
          .: {}
          f:backup-object-name: {}
          f:backup-object-uid: {}
      f:spec:
        .: {}
        f:pvcName: {}
        f:snapshotID: {}
      f:status: {}
    manager: px-backup
    operation: Update
    time: "2021-09-30T02:13:03Z"
  - apiVersion: kdmp.portworx.com/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:reason: {}
        f:status: {}
    manager: kopiaexecutor
    operation: Update
    time: "2021-09-30T02:13:23Z"
  name: d-dfb5046-2a9d986
  namespace: px-backup
  resourceVersion: "333778"
  selfLink: /apis/kdmp.portworx.com/v1alpha1/namespaces/px-backup/volumebackupdeletes/d-dfb5046-2a9d986
  uid: a737c546-2cdc-4635-8589-c5133beff0f2
spec:
  pvcName: nfs-data-cassandra-1
  snapshotID: 2a9d986e9ac5c288c05cd7f7ec5d75b3
status:
  reason: |
    repository [generic-backup/proxy-proxy/] connect failed: failed to run the kopia connect command: exit status 1 stdout:  stderr: kopia: error: error connecting to repository: repository not initialized in the provided storage, try --help
  status: Failed
[root@central-valiant-howler-0 ~]#
```

### CODE REVIEW

File: pkg/drivers/kopiadelete/kopiadelete.go [hunk 1]
```diff
@@ -202,19 +220,28 @@ func toJobName(jobName, snapshotID string) string {
 	if jobName != "" {
 		return jobName
 	}
-	return fmt.Sprintf("%s-%s", kopiaDeleteJobPrefix, snapshotID)
+	return fmt.Sprintf("%s%s", kopiaDeleteJobPrefix, snapshotID)
```
Comment (prashanthpx) [inline pkg/drivers/kopiadelete/kopiadelete.go:223]: Don't we need `-` here for separation?

File: pkg/executor/common.go [hunk 2]
```diff
@@ -379,6 +380,26 @@ func parseAzureCreds() (*Repository, error) {
...
+) error {
+	vd, err := kdmpschedops.Instance().GetVolumeBackupDelete(volumeBackupDeleteName, volumeBackupDeleteNamespace)
+	if err != nil {
+		return fmt.Errorf("failed in getting VolumeBackupDelete CR  [%s/%s]: %v", volumeBackupDeleteName, volumeBackupDeleteNamespace, err)
+	}
+	vd.Status.Status = statusType
+	vd.Status.Reason = errMsg
+
+	if _, err = kdmpschedops.Instance().UpdateVolumeBackupDelete(vd); err != nil {
+		return fmt.Errorf("failed in updating VolumeBackupDelte CR %s/%s VolumeDelete: %v", volumeBackupDeleteName, volumeBackupDeleteNamespace, err)
```
Comment (prashanthpx) [inline pkg/executor/common.go:398]: typo `VolumeBackupDelte`

**Review by prashanthpx — APPROVED** [review]

File: pkg/drivers/kopiadelete/kopiadelete.go (hunk 1 above)
Comment (siva-portworx) [reply pkg/drivers/kopiadelete/kopiadelete.go:223]: I modified the kopiaDeleteJobPrefix to hold the hypen as well. So I remove it here.

Comment (siva-portworx) [reply pkg/drivers/kopiadelete/kopiadelete.go]: Fixed.

File: pkg/executor/common.go (hunk 2 above)
Comment (siva-portworx) [reply pkg/executor/common.go]: Fixed

**Pull request merged**